1/contrib/s2i/run

```diff
@@ -12,6 +12,21 @@ source /usr/local/bin/kube-slave-common.sh
 jenkins_config_dir="/opt/openshift/configuration"
 jenkins_config_path="${jenkins_config_dir}/config.xml"
 
+
+CONTAINER_MEMORY_IN_BYTES=`cat /sys/fs/cgroup/memory/memory.limit_in_bytes`
+DEFAULT_MEMORY_CEILING=$((2**30-1))
+if [ "${CONTAINER_MEMORY_IN_BYTES}" -lt "${DEFAULT_MEMORY_CEILING}" ]; then
```

Comment:
(moving here so it's not a comment on an outdated diff).

I think we should always respect the memory limit the users chose, so capping at 1.5gigs (and forcing them to use a different mechanism) doesn't seem like the right thing to do to me.  If a user decides to run a jenkins container with a limit of 128gigs, we should assume that means they want ~128gigs of heap (w/ appropriate native memory reservations, as you're doing with the 80%). If they are doing something fancy where they want most of that memory for processes launched by jenkins, they can use the JAVA_OPTS override then.


The only case i'm particularly worried about is when they set no limit and therefore CONTAINER_MEMORY_IN_BYTES will be either "-1" or 9223372036854771712 and obviously using either of those to set a -Xmx value will be problematic (and if the value that represents "no limit" changes, any logic that uses those values to test for no limit will break).  

2**60 seems excessive, but 2**40 (~1 terabyte) is probably a safe upper limit.